README.md

```diff
@@ -82,8 +82,12 @@ Returns a new focus trap on `element` (one or more "containers" of tabbable node
 
 #### createOptions
 
-- **onActivate** `{() => void}`: A function that will be called when the focus trap activates.
-- **onDeactivate** `{() => void}`: A function that will be called when the focus trap deactivates,
+- **onActivate** `{() => void}`: A function that will be called **before** sending focus to the target element upon activation.
+- **onPostActivate** `{() => void}`: A function that will be called **after** sending focus to the target element upon activation.
+- **checkCanFocusTrap** `{(containers: Array<HTMLElement | SVGElement | string | () => HTMLElement | SVGElement>) => Promise<unknown>}`: Animated dialogs have a small delay between when `onActivate` is called and when the focus trap is focusable. `checkCanFocusTrap` expects a promise to be returned. When that promise resolves, focus will be sent to the focus-trap. Due to the lack of Promise support, `checkCanFocusTrap` is not supported in IE11 unless you provide a Promise polyfill.
+- **onDeactivate** `{() => void}`: A function that will be called **before** sending focus to the trigger element upon deactivation.
+- **onPostDeactivate** `{(value: unknown) => void}`: A function that will be called **after** sending focus to the trigger element upon deactivation.
+- **checkCanFocusTrigger** `{(trigger: HTMLElement | SVGElement) => Promise<unknown>}`: An animated trigger button will have a small delay between when `onDeactivate` is called and when the focus is sent back to the trigger. `checkCanFocusTrigger` expects a promise to be returned. When that promise resolves, focus will be sent to the trigger element. Due to the lack of Promise support, `checkCanFocusTrigger` is not supported in IE11 unless you provide a Promise polyfill.
```

Comment:
Also, I'd like to name this option, `checkCanReturnFocus`. That seems more appropriate to me for what it's gating, and the other options it relates to, those being `returnFocus`, `returnFocusOnDeactivate`, and `setReturnFocus`.

Reply:
Also, change `Promise<unknown>` to just `Promise`, and mention that focus will be returned to the configured node whether the promise resolves or rejects.

Reply:
I feel like sending focus back to the node on rejection is probably not a good idea. If the promise rejects then it probably means it isn't safe to return the node and so it might cause crashes or something if we try to return focus on rejection as well as success then it takes control away from developers. If the developer wants a promise that resolves even if something goes wrong, they can nest a promise inside a promise.

Reply:
I understand. The reason why I don't want to care what the Promise does is because I don't want to go too fast with this change, which could open a can of worms we're not thinking of right now. What should happen in this case or that case if it rejects? If we don't follow with setting focus, what state does that leave things in now, especially in the activation case? You have focus set on a node that's outside the already-activated trap, so right off the bat, focus has escaped. What happens when the user clicks the TAB key? Can of worms... 🙂 

For now, I'd like the Promise to solely be a _delay mechanism_ for what is the _inevitable_, that focus will be set (activate) or returned (deactivate), the same way it would be if the promise wasn't used.

As for the `finally`, the `then` signature is `then(resolvedHandler, rejectedHandler)`. That should be all that's necessary. What you show there (in this case, since there is only one promise and we don't care if `finishActivation()` throws or not) is functionally equivalent to `then(finishActivation, finishActivation)`.